### PR TITLE
Fix where log_task method fails when task is nil

### LINKED ISSUE #41: log_task method fails when task is nil
Here: https://github.com/tarantool/sharded-queue/blob/master/sharded_queue/api.lua#L46
When task is nil, error would be thrown since index of task tuple is accessed.
Since peek() method can be called with an id of a task that doesn't exist anymore, task would be nil in those cases.

### PR DESCRIPTION
**Pull request by naimulhaider**
Closes #41

### CODE REVIEW

File: sharded_queue/api.lua [hunk 1]
```diff
@@ -43,6 +43,11 @@ local function validate_options(options)
 end
 
 local function log_task(op_name, task)
+    if task == nil then
```
Comment (akudiyar) [inline sharded_queue/api.lua:46]: I'd rather not duplicate the log message and use more idiomatic `type(task) == 'table' and task[1] or task`. The expression may be put in a variable

**Review by akudiyar — CHANGES_REQUESTED** [review]

File: sharded_queue/api.lua (hunk 1 above)
Comment (akudiyar) [reply sharded_queue/api.lua:46]: Would mind checking other places with logging for the same behavior?

Comment (naimulhaider) [reply sharded_queue/api.lua:46]: Sure, in that case we have to change the formatter to '%s' since '%d' would throw error with nil values

Comment (naimulhaider) [reply sharded_queue/api.lua:46]: If you look at other places: https://github.com/tarantool/sharded-queue/blob/master/sharded_queue/drivers/fifottl.lua#L29

It's done pretty much like the way I did it.
Are you stil suggesting that we do this instead: `log.info(string.format([[Router[%d] %s: task %s]], fiber.self():id(), op_name, type(task) == 'table' and task[1] or task))`

Comment (akudiyar) [reply sharded_queue/api.lua:46]: I would like to decrease the verbosity and change other places as well. Btw for not making lines too long, putting the expression into a variable seems to look cleaner. Also all messages should follow the same format for nil tasks

File: sharded_queue/drivers/fifottl.lua [hunk 2]
```diff
@@ -27,12 +27,8 @@ local function is_expired(task)
...
 
 local function log_operation(op_name, task)
-    if task ~= nil then
-        log.info(("Storage[%d]: [%d] %s task %d")
-            :format(fiber.self():id(), task[index.bucket_id], op_name, task[index.task_id]))
-    else
-        log.info(("Storage[%d]: %s nil task"):format(fiber.self():id(), op_name))
-    end
+    local task_id = type(task) == 'table' and task[index.task_id] or task
+    log.info(string.format([[Storage[%d] %s: task %s]], fiber.self():id(), op_name, task_id))
```
Comment (akudiyar) [inline sharded_queue/drivers/fifottl.lua:31]: task[index.bucket_id] is missing. If the task is not table, let it be nil

**Review by akudiyar — CHANGES_REQUESTED** [review]

**Review by akudiyar — APPROVED** [review]

**Comment by akudiyar** [comment]
LGTM. Thank you for the patch!

**Pull request merged**